### DISCUSSION TITLE
Show and Tell: Find and highlight the MCS (maximum common substructure) between a set of molecules

### DISCUSSION THREAD

**Discussion by bertiewooster** [rdkit/rdkit#5751] [Show and tell]
Hi, I made a utility function that visually displays the [maximum common substructure between a set of molecules using RDKit](https://bertiewooster.github.io/2022/10/09/RDKit-find-and-highlight-the-maximum-common-substructure-between-molecules.html). You give it the SMILES of a set of molecules, and It finds the MCS as SMARTS, displays the MCS as a molecule, and displays the molecules with the MCS highlighted, aligning the MCS by default:

`SmilesMCStoGridImage({"NC1OC1": "amine", "C1OC1[N+](=O)[O-]": "nitro"})`

produces

![2022-10-09-RDKit-find-and-highlight-the-maximum-common-substructure-between-molecules_11_0](https://user-images.githubusercontent.com/1335522/201432201-2f12d56e-0314-4d9d-9f26-210293d3d5b6.png)

Code is available in that [blog post](https://bertiewooster.github.io/2022/10/09/RDKit-find-and-highlight-the-maximum-common-substructure-between-molecules.html) and accompanying Jupyter notebook.

**Comment by greglandrum** [comment]
Thanks for sharing that @bertiewooster !

**Reply by bertiewooster** [reply]
Glad you like it @greglandrum! I'm enjoying visualizing RDKit's powerful capabilities.
